src/multitaper.jl

```diff
@@ -0,0 +1,518 @@
+                                signal::AbstractVector,
+                                config::MTSpectrogramConfig) -> Spectrogram
+
+Computes a multitaper spectrogram using the parameters specified in `config`.
+
+* `destination`: `config.mt_config.n_freq_bins` x `config.n_time_points` matrix. This can be created by `allocate_output(config)`.
+* `signal`: vector of length `config.n_samples`
+* `config`: an [`MTSpectrogramConfig`](@ref) object to hold temporary variables and configuration settings.
+"""
+@views function mt_spectrogram!(destination::AbstractMatrix,
+                                          signal::AbstractVector,
+                                          config::MTSpectrogramConfig)
+    if size(destination) != (config.mt_config.n_freq_bins, config.n_time_points)
+        throw(DimensionMismatch("""Expected `destination` to be of size `(config.mt_config.n_freq_bins, config.n_time_points)`;
+                got `size(destination) == $(size(destination)) and
+                `(config.mt_config.n_freq_bins, config.n_time_points)` = $((config.mt_config.n_freq_bins, config.n_time_points))"""))
+    end
+    if length(signal) != config.n_samples
+        throw(DimensionMismatch("""Expected `signal` to be of length `config.n_samples`;
+                got `length(signal) = $(length(signal)) and `config.n_samples` = $(config.n_samples)"""))
+    end
+    samples_per_window = config.mt_config.n_samples
+    subepochs = arraysplit(signal, samples_per_window, config.n_overlap_samples)
+    for (time_index, subepoch) in enumerate(subepochs)
+        mt_pgram!(destination[:, time_index], subepoch, config.mt_config)
+    end
+    return Spectrogram(destination, config.mt_config.freq, config.time)
+end
+
+allocate_output(config::MTSpectrogramConfig{T}) where {T} = Matrix{fftabs2type(T)}(undef, config.mt_config.n_freq_bins, config.n_time_points)
+
+"""
+    mt_spectrogram(signal::AbstractVector{T}, n::Int=length(s) >> 3,
+                                  n_overlap::Int=n >> 1; fs::Int=1,
+                                  onesided::Bool=T <: Real, kwargs...) where {T} -> Spectrogram
+
+Compute a multitaper spectrogram. Any additional keyword arguments accepted by [`MTConfig`](@ref)
+may be passed to configure the tapering. See also [`mt_spectrogram!`](@ref).
+"""
+function mt_spectrogram(signal::AbstractVector{T}, n::Int=length(signal) >> 3,
+                                  n_overlap::Int=n >> 1; fs::Int=1,
+                                  onesided::Bool=T <: Real, kwargs...) where {T}
+    config = MTSpectrogramConfig{T}(length(signal), n, n_overlap; fs=fs,
+                                              onesided=onesided, fft_flags = FFTW.ESTIMATE, kwargs...)
+    X = allocate_output(config)
+    return mt_spectrogram!(X, signal, config)
+end
+
+#####
+##### Multitapered cross spectral matrix
+#####
+
+struct CrossSpectral{T, F}
+    values::T
+    freq::F
+end
+
+freq(c::CrossSpectral) = c.freq
```

Comment:
no helper for the `values`, so I guess the user should just use-field access. Open to adding something if there's a name that makes sense

Reply:
done in the latest commit